### PR TITLE
Search optimisations

### PR DESCRIPTION
**Pull request by eddturner**

### CODE REVIEW

**Review by ahmadshadab — COMMENTED** [review]
Wow, this commit has handled many cases, complicated work. Good job. Hope(the comprehensive tests give me confidence), it doesn't break anything.

File: integration-test/src/test/java/org/uniprot/store/indexer/search/uniprot/FullCISplitNumericAnalysisSearchIT.java [hunk 1]
```diff
@@ -0,0 +1,802 @@
+package org.uniprot.store.indexer.search.uniprot;
```
[the remark this answers is not in the document]
Comment (eddturner) [reply integration-test/src/test/java/org/uniprot/store/indexer/search/uniprot/FullCISplitNumericAnalysisSearchIT.java:1]: I know what you mean about the size of the class, 90% of its size is just the different test cases ... and they don't seem different enough to refactor ...

We left the synonym test in as disabled as we are anticipating someone saying they want synonyms available (which are handled on uniprot.org).

File: indexer/src/main/java/org/uniprot/store/indexer/uniprotkb/processor/InactiveEntryConverter.java [hunk 2]
```diff
@@ -20,12 +20,14 @@ public UniProtDocument convert(InactiveUniProtEntry source) {
...
         }
-        if (source.getReason().equalsIgnoreCase("deleted")) {
-            document.content.add(source.getAccession());
-        }
+        // TODO: 17/12/2021 no need to add accession to content when already added to accession
+        // field in document.
+        //        if (source.getReason().equalsIgnoreCase("deleted")) {
+        //            document.content.add(source.getAccession());
+        //        }
         document.inactiveReason = source.getInactiveReason();
```
[the remark this answers is not in the document]
Comment (eddturner) [reply indexer/src/main/java/org/uniprot/store/indexer/uniprotkb/processor/InactiveEntryConverter.java:31]: woops. done

File: spark-indexer/src/main/java/org/uniprot/store/spark/indexer/uniprot/converter/InactiveUniprotEntryConverter.java [hunk 3]
```diff
@@ -24,14 +24,20 @@ public UniProtDocument convert(UniProtKBEntry source) {
...
-        document.inactiveReason = type.getDisplayName();
+//        if (type.equals(InactiveReasonType.DELETED)) {
+//            // todo why adding to content? i think it'd be better to add to accession
+//            // content is a field added to default fields (qf), and should only be used for
+//            // dynamic fields (because we can't add all dynamic fields to qf (too many).
+//            document.content.add(source.getPrimaryAccession().getValue());
+//
+//            // TODO: 17/12/2021 Question: what does adding it to content do that is different from adding
+//            // it to document.accession (line 23)?
+//        }
```
[the remark this answers is not in the document]
Comment (eddturner) [reply spark-indexer/src/main/java/org/uniprot/store/spark/indexer/uniprot/converter/InactiveUniprotEntryConverter.java:39]: ta

File: integration-test/src/test/java/org/uniprot/store/indexer/search/uniprot/FullCISplitNumericAnalysisSearchIT.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (eddturner) [reply integration-test/src/test/java/org/uniprot/store/indexer/search/uniprot/FullCISplitNumericAnalysisSearchIT.java:1]: I'm going to create a jira about refactoring these classes (I really only added 1 test to it in this issue).

[the remark this answers is not in the document]
Comment (eddturner) [reply integration-test/src/test/java/org/uniprot/store/indexer/search/uniprot/FullCISplitNumericAnalysisSearchIT.java:1]: https://www.ebi.ac.uk/panda/jira/browse/TRM-27227

**Pull request merged**